core/contract/src/test/java/org/eclipse/dataspaceconnector/contract/negotiation/ContractNegotiationIntegrationTest.java

```diff
@@ -157,24 +178,28 @@ void testNegotiation_agreementDeclined() throws Exception {
                 .build();
         consumerManager.initiate(request);
 
-        // Wait for negotiation to finish with time out at 15 seconds
-        var success = countDownLatch.await(15, TimeUnit.SECONDS);
-
-        assertThat(success).isTrue();
-
-        var consumerNegotiation = consumerStore.find(consumerNegotiationId);
-        var providerNegotiation = providerStore.findForCorrelationId(consumerNegotiationId);
-
-        // Assert that provider and consumer have the same offers stored
-        assertThat(consumerNegotiation.getContractOffers()).hasSize(1);
-        assertThat(consumerNegotiation.getContractOffers().size()).isEqualTo(providerNegotiation.getContractOffers().size());
-        assertThat(consumerNegotiation.getLastContractOffer()).isEqualTo(providerNegotiation.getLastContractOffer());
-
-        // Assert that no agreement has been stored on either side
-        assertThat(consumerNegotiation.getContractAgreement()).isNull();
-        assertThat(providerNegotiation.getContractAgreement()).isNull();
-        verify(validationService, Mockito.atLeastOnce()).validate(token, offer);
-        verify(validationService, Mockito.atLeastOnce()).validate(eq(token), any(ContractAgreement.class), any(ContractOffer.class));
+        await().atMost(DEFAULT_TEST_TIMEOUT)
+                .pollInterval(DEFAULT_POLL_INTERVAL)
+                .untilAsserted(() -> {
+                    var consumerNegotiation = consumerStore.find(consumerNegotiationId);
+                    var providerNegotiation = providerStore.findForCorrelationId(consumerNegotiationId);
+
+                    // Assert that provider and consumer have the same offers stored
+                    assertThat(consumerNegotiation).isNotNull();
```

Comment:
seems much duplicate code?

Reply:
i extracted a method that performs some basic assertions, but I'm not sure whether that makes the code _more_ or _less_ readable.